lib/engine-jit/src/serialize.rs

```diff
@@ -38,9 +31,91 @@ pub struct SerializableCompilation {
 
 /// Serializable struct that is able to serialize from and to
 /// a `JITArtifactInfo`.
-#[derive(Serialize, Deserialize, MemoryUsage)]
+#[derive(MemoryUsage, Archive, RkyvDeserialize, RkyvSerialize)]
 pub struct SerializableModule {
     pub compilation: SerializableCompilation,
     pub compile_info: CompileModuleInfo,
     pub data_initializers: Box<[OwnedDataInitializer]>,
 }
+
+fn to_serialize_error(err: impl std::error::Error) -> SerializeError {
+    SerializeError::Generic(format!("{}", err))
+}
+
+cfg_if::cfg_if! {
+    if #[cfg(target_endian = "big")] {
+        const HOST_ENDIAN: u8 = b'b';
+    } else if #[cfg(target_endian = "little")] {
+        const HOST_ENDIAN: u8 = b'l';
+    }
+    else {
+        compile_error!("Endian not supported by the host");
+    }
+}
+
+impl SerializableModule {
+    /// Serialize a Module into bytes
+    /// The bytes will have the following format:
+    /// RKYV serialization (any length) + POS (8 bytes) + Endian (1 byte)
+    pub fn serialize(&self) -> Result<Vec<u8>, SerializeError> {
+        let mut serializer = SharedSerializerAdapter::new(WriteSerializer::new(vec![]));
+        let pos = serializer
+            .serialize_value(self)
+            .map_err(to_serialize_error)? as u64;
+        let mut serialized_data = serializer.into_inner().into_inner();
+        serialized_data.extend_from_slice(&pos.to_le_bytes());
+        serialized_data.extend_from_slice(&[HOST_ENDIAN]);
+        Ok(serialized_data)
+    }
+
+    /// Deserialize a Module from a slice.
+    /// The slice must have the following format:
+    /// RKYV serialization (any length) + POS (8 bytes) + Endian (1 byte)
+    ///
+    /// # Safety
+    ///
+    /// This method is unsafe since it deserializes data directly
+    /// from memory.
+    /// Right now we are not doing any extra work for validation, but
+    /// `rkyv` has an option to do bytecheck on the serialized data before
+    /// serializing (via `rkyv::check_archived_value`).
+    pub unsafe fn deserialize(metadata_slice: &[u8]) -> Result<Self, DeserializeError> {
+        let archived = Self::archive_from_slice(metadata_slice)?;
+        Self::deserialize_from_archive(archived)
+    }
+
+    unsafe fn archive_from_slice<'a>(
```

Comment:
Still need a safety section on this (see https://github.com/wasmerio/wasmer/pull/2250/files#r623939328)

Reply:
Fixed!